server/index.ts

```diff
@@ -216,6 +233,31 @@ app.all(
 		: getRequestHandler(build),
 )
 
+const metricsApp = express()
+
+metricsApp.use(morgan('tiny'))
+
+metricsApp.get('/metrics', async (_, res) => {
+	const metrics = await prometheus.client.register.metrics()
+	res.set('Content-Type', 'text/plain')
+	res.send(metrics)
+})
+
+const desiredMetricsPort = 9091
+const metricsPort = await getPort({
+	port: portNumbers(desiredMetricsPort, desiredMetricsPort + 100),
+})
+const metricsServer = metricsApp.listen(metricsPort, () => {
+	if (metricsPort !== desiredMetricsPort) {
+		console.warn(
+			chalk.yellow(
+				`⚠️  Metrics port ${desiredMetricsPort} is not available, using ${metricsPort} instead.`,
+			),
+		)
+	}
+	console.log(`📈 Metrics server listening on port ${metricsPort}`)
+})
```

Comment:
Should we only enable this in production somehow? I'm thinking an environment variable `ENABLE_METRICS` or something. Most of the time local development really doesn't need the metrics.

Reply:
Added a environment variable, and checked it and production before launching the metrics server.